Dockerfile

```diff
@@ -1,71 +1,54 @@
-# Build compilation image
-FROM ruby:3.1.4-alpine as builder
+FROM ruby:3.1.4-alpine3.18 as base
 
-# The application runs from /app
 WORKDIR /app
 
-# Add the timezone as it's not configured by default in Alpine
+# Add the timezone as it’s not configured by default in Alpine
 RUN apk add --update --no-cache tzdata && \
   cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
   echo "Europe/London" > /etc/timezone
 
-# build-base: complication tools for bundle
+# build-base: compilation tools for bundle
 # yarn: node package manager
 # postgresql-dev: postgres driver and libraries
-RUN apk add --no-cache build-base yarn postgresql-dev git
+RUN apk add --no-cache build-base yarn postgresql-dev git bash
 
-# Install bundler to run bundle exec
-# This should be the same version as the Gemfile.lock
+# Bundler version should be the same version as what the Gemfile.lock was bundled with
 RUN gem install bundler:2.3.14 --no-document
 
-# Install gems defined in Gemfile
 COPY .ruby-version Gemfile Gemfile.lock /app/
+RUN bundle config set without "development"
+RUN bundle install --jobs=4 --no-binstubs --no-cache
 
-RUN bundle config set without "development test"
-ARG BUNDLE_FLAGS="--jobs=4 --no-binstubs --no-cache"
-RUN bundle install ${BUNDLE_FLAGS}
-
-# Install node packages defined in package.json, including webpack
 COPY package.json yarn.lock /app/
 RUN yarn install --frozen-lockfile
 
-# Copy all files to /app (except what is defined in .dockerignore)
 COPY . /app/
 
-# Compile assets and run webpack. We set a dummy secret key.
-RUN RAILS_ENV=production bundle exec rails SECRET_KEY_BASE=pickasecuretoken assets:precompile
+RUN bundle exec rake assets:precompile
 
-# Cleanup to save space in the production image
-RUN rm -rf node_modules log tmp && \
-      rm -rf /usr/local/bundle/cache && \
-      rm -rf .env && \
-      find /usr/local/bundle/gems -name "*.c" -delete && \
-      find /usr/local/bundle/gems -name "*.h" -delete && \
-      find /usr/local/bundle/gems -name "*.o" -delete && \
-      find /usr/local/bundle/gems -name "*.html" -delete
+ENV PORT=8080
 
-# Build runtime image
-FROM ruby:3.1.4-alpine as production
+EXPOSE ${PORT}
 
-# The application runs from /app
-WORKDIR /app
+FROM base as development
 
-# Add postgres driver library
-# Add the timezone as it's not configured by default in Alpine
-RUN apk add --update --no-cache libpq tzdata && \
-  cp /usr/share/zoneinfo/Europe/London /etc/localtime && \
-  echo "Europe/London" > /etc/timezone
+RUN bundle config set without ""
```

Comment:
Since the `without` command persists what was set before (in this case `without development` further up in the Dockerfile), we need to set it to blank so it installs the "development" group gems here